cmd/mirroring/main.go

```diff
@@ -237,6 +259,36 @@ func main() {
 			log.Fatalf("failed to delete old checkpoints: %v", err)
 		}
 
+		// Search for identities in the log range
+		if len(ids.Identities) > 0 {
+			entries, err := rekor.GetEntriesByIndexRange(context.Background(), rekorClient, startIndex, endIndex)
+			if err != nil {
+				log.Fatalf("error getting entries by index range: %v", err)
+			}
+			idEntries, err := rekor.MatchedIndices(entries, ids)
+			if err != nil {
+				log.Fatalf("error finding log indices: %v", err)
+			}
+
+			if len(idEntries) > 0 {
+				for _, idEntry := range idEntries {
+					log.Printf("Found subject %s, issuer %s at log index %d, uuid %s\n",
+						idEntry.Subject, idEntry.Issuer, idEntry.Index, idEntry.UUID)
+
+					idFile, err := os.OpenFile(*identitiesFoundFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+					if err != nil {
+						idFile.Close()
+						log.Fatalf("failed to open identities file: %v", err)
+					}
+					if _, err := idFile.WriteString(fmt.Sprintf("%s %s %d %s\n", idEntry.Subject, idEntry.Issuer, idEntry.Index, idEntry.UUID)); err != nil {
+						idFile.Close()
+						log.Fatalf("failed to write to file: %v", err)
+					}
+					idFile.Close()
```

Comment:
make this a defer statement instead of putting before every exit

Reply:
Linter complains about log.Fatalf occurring before the file closes unfortunately - The right fix is to not call log.fatalf and fail gracefully. I'll file an issue